src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -0,0 +1,200 @@
+package io.pinecone.helpers;
+
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeException;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import org.openapitools.client.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.pinecone.helpers.BuildUpsertRequest.buildRequiredUpsertRequestByDimension;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class TestIndexResourcesManager {
+    private static final Logger logger = LoggerFactory.getLogger(IndexManager.class);
+    private static TestIndexResourcesManager instance;
+    private static final String apiKey = System.getenv("PINECONE_API_KEY");
+    private final int dimension = System.getenv("DIMENSION") == null
+            ? 4
+            : Integer.parseInt(System.getenv("DIMENSION"));
+    private final String environment = System.getenv("PINECONE_ENVIRONMENT") == null
+            ? "us-east4-gcp"
+            : System.getenv("PINECONE_ENVIRONMENT");
+    private static final IndexMetric metric = System.getenv("METRIC") == null
+            ? IndexMetric.COSINE
+            : IndexMetric.valueOf(System.getenv("METRIC"));
+    private Pinecone pineconeClient;
+    private String podIndexName;
+    private IndexModel podIndexModel;
+    private String serverlessIndexName;
+    private IndexModel serverlessIndexModel;
+    private String collectionName;
+    private CollectionModel collectionModel;
+    private final List<String> podIndexVectorIds = Arrays.asList("v1", "v2", "v3");
```

Comment:
Total nit, which I know I'll likely get pushback on because I think `v1` etc is how we represent vector IDs in other parts of the system and documentation, but I find it more intuitive to read if IDs are like `id1`. IMO `v1` seems more like an object, like the actual "vector 1".

Reply:
I think it's a bit arbitrary what the ids are, I can change them up. 👍